### PR TITLE
Kotlin migration SyncErrorDialog

### PR DESCRIPTION
**Pull request by Arnold2381**
Kotlin migration SyncErrorDialog

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt [hunk 1]
```diff
@@ -14,276 +14,262 @@
...
+        fun mediaCheck()
+        fun dismissAllDialogFragments()
+        fun integrityCheck()
     }
 
+    override fun onCreateDialog(savedInstanceState: Bundle?): MaterialDialog {
+        super.onCreate(savedInstanceState)
+        val builder = MaterialDialog.Builder(requireActivity())
+            .title(title)
+            .content(message!!)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt:47]: !

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt [hunk 2]
```diff
@@ -14,276 +14,262 @@
...
+                    .negativeText(R.string.dialog_cancel)
+                    .onPositive { _: MaterialDialog?, _: DialogAction? ->
+                        (activity as SyncErrorDialogListener?)!!.integrityCheck()
+                        dismissAllDialogFragments()
+                    }
+                    .show()
             }
-            default:
-                return null;
+            else -> null!!
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt:182]: This fails, but I feel it's fine as it's a logic error.

Maybe follow-up to use an enum so this won't happen

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt [hunk 3]
```diff
@@ -14,276 +14,262 @@
...
+     *
      * @return message to be shown in notification in bar
      */
-    @Override
-    public String getNotificationMessage() {
-        if (getArguments().getInt("dialogType") == DIALOG_USER_NOT_LOGGED_IN_SYNC) {
-            return res().getString(R.string.not_logged_in_title);
-        }
-        return getMessage();
+    override fun getNotificationMessage(): String {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt:226]: I think this is nullable in the base class (we might want to Kotlin convert the base class, or at least add a nullability reference here

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt (hunk 3 above)
Comment (Arnold2381) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt:226]: is it needed here? because it always return non null value.

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt:227]: I believe `this.message` https://github.com/ankidroid/Anki-Android/pull/9833/files/d5dabac6c0e21a43456b987ad670820da9300522..3992c8a9b23cbb13b64e0dbe6495f5e6922d8bdf#diff-6d27b6a54b50b4dc28b250dd9461251a564e99a5e1c0aafea2ef5cf45095a35bR218 is nullable

**Review by david-allison-1 — COMMENTED** [review]
Forgot to press "Submit"

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**